### PR TITLE
Add rebuild support to repo toolset

### PR DESCRIPTION
**Pull request by 333fred**
Allows users to specify doing a full rebuild. This change is split into two commits, one for the gitignore and whitespace fixes, and one for the actual content change. Tagging @tmat for review.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -14,6 +14,7 @@
 *.nuget.props
 *.nuget.targets
 NuGet.exe
+nuget.exe
```
Comment (tmat) [inline .gitignore]: nuget.exe shouldn't be used at all. The toolset uses msbuild or dotnet cli to restore. How does nuget show up?

Comment (333fred) [inline .gitignore]: https://github.com/dotnet/roslyn-tools/blob/master/build/Scripts/Windows/Build.ps1#L153. Really, I should change this to be [Nn]u[Gg]et.exe

Comment (333fred) [inline .gitignore]: Because I use bash, and git for linux treats NuGet.exe and nuget.exe as different files.

Comment (tmat) [inline .gitignore]: Oh, I see. This repo itself doesn't use RepoToolset (it should!). Looks good.

**Pull request merged**